docs/translations/translations-in-JS-TS-files.md

```diff
@@ -0,0 +1,108 @@
+# Translations in JS/TS files
+
+In comparison to PHP files, translations in JS/TS files require a few additional steps. First, the Node package [`@wordpress/i18n`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/) needs to be installed:
+
+```sh
+npm install @wordpress/i18n --save
+```
+
+Next, each JS/TS file, that hold strings which should be translated, needs to import the corresponding functions from `@wordpress/i18n`:
+
+```ts
+import { sprintf, _n } from '@wordpress/i18n';
+```
+
+## Usage of localization functions
+
+### `__()`
+
+The function `__()` retrieves the translation of `$text`.
+
+```ts
+// Schema
+const translation = __( string text, string domain = 'default' );
+
+// Example
+import { __ } from '@wordpress/i18n';
+
+const translation = __( 'Place Order', 'woo-gutenberg-products-block' );
```

Comment:
I'm not sure what you mean by that. Can you give me an example, @senadir?

Reply:
Writing translation like this won't work:

```js
const translation = __( someVariable, 'woo-gutenberg-products-block' );
```

Because translation extraction plugins work on static files and can't evaluate code, so `someVariable` will never be evaluated and extracted.

Reply:
> Writing translation like this won't work:
> 
> ```js
> const translation = __( someVariable, 'woo-gutenberg-products-block' );
> ```
> 
> Because translation extraction plugins work on static files and can't evaluate code, so `someVariable` will never be evaluated and extracted.

While I see your point, I don't see how this is related to translations given that `someVariable` is a variable and not a string. 

The examples explicitly mention that a string is expected, e.g.

```ts
// Schema
const translation = __( string text, string domain = 'default' );

// Example
import { __ } from '@wordpress/i18n';

const translation = __( 'Place Order', 'woo-gutenberg-products-block' );
```

Reply:
> Not sure if this is common, but people may use template literals here, and it won't work as well.
> 
> ```
> const translation = __( `Your cart (${count} items`, 'woo-gutenberg-products-block' );
> ```

I've added a section about `sprintf()` to the _Translations in JS/TS files_ doc, to highlight that template literals cannot be used in translation functions.

Reply:
There's nothing there that explicitly mentions it has to be a primitive string, not a variable pointing to a string, which is still an issue that a lot of people fall into to the point of creating a linter rule for it.
https://github.com/WordPress/gutenberg/blob/trunk/packages/eslint-plugin/docs/rules/i18n-no-variables.md

Reply:
Now I see what you meant. I've added further information on the section I mentioned before.